### PR TITLE
STY/FW: Remove convenience methods imported from pandas, import datetime from datetime

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #366.

Starting with pandas 1.0.0, `pandas.datetime` is deprecated.  Updating pysat 3.0.0 to import directly from datetime, not pandas, as recommended by pandas.  Also removes the convenience import of objects from `pandas`, including the deprecated `pandas.Panel`.

**Update: leaving in datetime.datetime and pandas.DataFrame as convenience methods for the time being to prevent breaking changes in downstream packages.  Merging now as changes are extensive here.**

## Type of change

- Future Warning fix (non-breaking change which fixes an issue)
- Breaking change (Panel is gone!)

# How Has This Been Tested?

Tested in Travis CI with pandas 1.0.0.  All tests pass, no future warnings.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
NOTE: removal of pysat.DataFrame convenience method here requires similar upgrade in pysatCDF.

**Comment by jklenzing** [comment]
Requires pysat/pysatCDF#22 merge to complete

**Comment by jklenzing** [comment]
Just updated this with the latest commits from develop-3.  Still need to update to use the latest pysatCDF changes before testing and review.  Two possible paths: I need a pip-installable version of pysatCDF develop (ie, a patch release), or I can update Travis here to install from github and switch to develop.  @rstoneback?

**Comment by rstoneback** [comment]
Updating travis is almost guaranteed to be faster.

**Comment by jklenzing** [comment]
develop ==> 1227 passed, 5683 warnings in 1240.38 seconds 
this branch ==> 1227 passed, 2854 warnings in 1156.19 seconds

**Comment by rstoneback** [comment]
Nice.

On Apr 2, 2020, at 4:01 PM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:



develop ==> 1227 passed, 5683 warnings in 1240.38 seconds
this branch ==> 1227 passed, 2854 warnings in 1156.19 seconds

File: demo/cnofs_vefi_dc_b_orbit_plots.py [hunk 1]
```diff
@@ -4,9 +4,12 @@
 """
 
 import os
-import pysat
+import datetime as dt
+
```
Comment (aburrell) [inline demo/cnofs_vefi_dc_b_orbit_plots.py:8]: Remove extra whitespace

File: docs/examples.rst [hunk 2]
```diff
@@ -10,6 +10,7 @@ Plotting a series of orbit-by-orbit plots is a great way to become familiar with
 
 .. code:: python
 
+   import datetime as dt
    import os
    import pysat
    import matplotlib.pyplot as plt
```
Comment (aburrell) [inline docs/examples.rst:16]: Change import order

File: pysat/_files.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import weakref
 import re
 import glob
+import datetime as dt
 import numpy as np
 import pandas as pds
 from pysat import data_dir as data_dir
```
Comment (aburrell) [inline pysat/_files.py:12]: Fix import order.

File: pysat/instruments/champ_star.py [hunk 4]
```diff
@@ -153,7 +154,6 @@ def load(fnames, tag=None, sat_id=None):
     def parse_champdate(y, d, s):
         '''parse CHAMP date string (YY DDD SSSSS) into a datetime object
         '''
```
Comment (aburrell) [inline pysat/instruments/champ_star.py:156]: Need to replace these with `"""`

File: pysat/instruments/superdarn_grdex.py [hunk 5]
```diff
@@ -33,6 +33,7 @@
 
 from __future__ import print_function
 from __future__ import absolute_import
+import datetime as dt
 import sys
 import os
 import functools
```
Comment (aburrell) [inline pysat/instruments/superdarn_grdex.py:39]: import order

File: pysat/instruments/sw_dst.py [hunk 6]
```diff
@@ -27,6 +27,7 @@
 """
 
 import os
+import datetime as dt
 import pandas as pds
 import numpy as np
```
Comment (aburrell) [inline pysat/instruments/sw_dst.py:32]: Import order

File: pysat/instruments/templates/netcdf_pandas.py [hunk 7]
```diff
@@ -22,8 +22,7 @@
 
 """
 
-import pandas as pds
-import numpy as np
+import datetime as dt
 import pysat
 
 import logging
```
Comment (aburrell) [inline pysat/instruments/templates/netcdf_pandas.py:28]: logging belongs under datetime

File: pysat/instruments/templates/netcdf_pandas.py [hunk 8]
```diff
@@ -211,5 +210,5 @@ def download(date_array, tag, sat_id, data_path=None, user=None,
     """
 
     logger.info('This is a generic Instrument routine and does not support ' +
-          'downloading data.')
+                'downloading data.')
```
Comment (aburrell) [inline pysat/instruments/templates/netcdf_pandas.py:213]: Our style recommends using 'join'

File: pysat/instruments/templates/template_cdaweb_instrument.py [hunk 9]
```diff
@@ -39,10 +39,7 @@
 from __future__ import absolute_import
 
 import functools
-import numpy as np
-import sys
-import pandas as pds
-import pysat
+import datetime as dt
```
Comment (aburrell) [inline pysat/instruments/templates/template_cdaweb_instrument.py:42]: not alphabetical

File: pysat/instruments/templates/template_instrument.py [hunk 10]
```diff
@@ -39,6 +39,7 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
+import datetime as dt
 # pandas support
 import pandas as pds
 # xarray support
```
Comment (aburrell) [inline pysat/instruments/templates/template_instrument.py:45]: Do we need these comments?

File: pysat/tests/test_instrument.py [hunk 11]
```diff
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 # Test some of the basic _core functions
 import numpy as np
+import datetime as dt
 import sys
```
Comment (aburrell) [inline pysat/tests/test_instrument.py:5]: fix order

File: pysat/tests/test_orbits.py [hunk 12]
```diff
@@ -1,5 +1,6 @@
 import datetime as dt
 from dateutil.relativedelta import relativedelta as relativedelta
+import datetime as dt
```
Comment (aburrell) [inline pysat/tests/test_orbits.py:3]: Too many datetime imports

File: pysat/tests/test_utils.py [hunk 13]
```diff
@@ -3,10 +3,8 @@
 """
 import os
 import tempfile
-import warnings
 from nose.tools import assert_raises, raises
 import numpy as np
```
Comment (aburrell) [inline pysat/tests/test_utils.py:7]: Import order

File: pysat/tests/test_utils_time.py [hunk 14]
```diff
@@ -4,7 +4,7 @@
 import numpy as np
 
 from nose.tools import raises
-import pandas as pds
+import datetime as dt
```
Comment (aburrell) [inline pysat/tests/test_utils_time.py:7]: import order and whitespace

**Comment by jklenzing** [comment]
Thanks @aburrell.  Updates to style throughout.

**Comment by jklenzing** [comment]
TODO before merging:
- [ ] merge pysat/pysatModels#39
- [ ] merge pysat/pysatMissions#30
- [ ] merge pysat/pysatSeasons#17 (affects demo codes there, not unit tests)
- [ ] pip version of pysatCDF develop or replacement for pysatCDF

File: pysat/instruments/templates/template_instrument.py (hunk 10 above)
Comment (jklenzing) [reply pysat/instruments/templates/template_instrument.py]: no

File: pysat/tests/test_instrument.py [hunk 15]
```diff
@@ -45,9 +46,9 @@ def test_basic_instrument_load(self):
         data)."""
         self.testInst.load(2009, 1)
         test_date = self.testInst.index[0]
-        test_date = pysat.datetime(test_date.year, test_date.month,
-                                   test_date.day)
-        assert (test_date == pds.datetime(2009, 1, 1)) & \
+        test_date = dt.datetime(test_date.year, test_date.month,
+                                test_date.day)
+        assert (test_date == dt.datetime(2009, 1, 1)) & \
```
Comment (aburrell) [inline pysat/tests/test_instrument.py:51]: flake8 will want the operator at the beginning of the next line.

File: pysat/tests/test_instrument.py [hunk 16]
```diff
@@ -76,98 +77,98 @@ def test_basic_instrument_load_by_file_and_multifile(self):
...
     def test_basic_instrument_load_leap_year(self):
         """Test if the correct day is being loaded (Leap-Year)."""
         self.testInst.load(2008, 366)
         test_date = self.testInst.index[0]
-        test_date = pysat.datetime(test_date.year, test_date.month,
-                                   test_date.day)
-        assert (test_date == pds.datetime(2008, 12, 31)) & \
+        test_date = dt.datetime(test_date.year, test_date.month,
+                                test_date.day)
+        assert (test_date == dt.datetime(2008, 12, 31)) & \
```
Comment (aburrell) [inline pysat/tests/test_instrument.py:110]: Same here.

File: pysat/tests/test_instrument.py [hunk 17]
```diff
@@ -76,98 +77,98 @@ def test_basic_instrument_load_by_file_and_multifile(self):
...
     def test_basic_fid_instrument_load(self):
         """Test if first day is loaded by default when first invoking .next."""
         self.testInst.load(fid=0)
         test_date = self.testInst.index[0]
-        test_date = pysat.datetime(test_date.year, test_date.month,
-                                   test_date.day)
-        assert (test_date == pds.datetime(2008, 1, 1)) & \
+        test_date = dt.datetime(test_date.year, test_date.month,
+                                test_date.day)
+        assert (test_date == dt.datetime(2008, 1, 1)) & \
```
Comment (aburrell) [inline pysat/tests/test_instrument.py:135]: And here

Comment (aburrell) [reply pysat/tests/test_instrument.py:135]: And places below.

File: pysat/tests/test_instrument.py (hunk 15 above)
Comment (jklenzing) [reply pysat/tests/test_instrument.py:51]: In other places, I've been rewriting these as separate assert statements.  Could do that here.

**Comment by jklenzing** [comment]
Updated to address AGB comments on flake8 conventions.  This still breaks due to pysatCDF, so converting to a draft until that is sorted out.

**Comment by jklenzing** [comment]
After merging develop-3 into this branch, I realized we should probably sort this out sooner rather than later or it goes stale quickly.  I've added the convenience methods for datetime and DataFrame back in temporarily so this is no longer a breaking change, but this code does not use those methods at all.  Instead it imports (or at least should) directly from datetime and pandas, respectively, whenever needed.  We can remove the convenience methods in a later pull request

This will fix roughly half the warning messages.

**Comment by rstoneback** [comment]
New version of pysatCDF has been released. Source code and delocated builds for 4 different python versions on Mac OS have been uploaded to pypi.

**Comment by jklenzing** [comment]
Thanks @rstoneback.  I'll update once the next round of testing updates are done.

**Pull request merged**